solutions/devsprint-firas-safa-1/GitHubApp/Data/Network/NetworkManager.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  NetworkEngine.swift
+//  GitHubApp
+//
+//  Created by Razee Hussein-Jamal on 09/02/2022.
+//
+
+import Foundation
+
+protocol NetworkRequestProtocol {
+    func request<ResponseType: Decodable>(with requestType: HTTPRequest) async throws -> [ResponseType]
```

Comment:
The return value should not be an array of `ResponseType` but only `ResponseType`
When you make your request, and imply your return value, that return value should be the array of response type

// let response1: [ResponseType] = manager.request(REQUEST())
// let response1: ResponseType = manager.request(REQUEST())